src/platform/web/lib/util/http.ts

```diff
@@ -71,17 +80,27 @@ const Http: typeof IHttp = class {
       };
 
       this.checkConnectivity = function (callback: (err: ErrorInfo | null, connectivity: boolean) => void) {
-        const upUrl = Defaults.internetUpUrl;
-        Logger.logAction(Logger.LOG_MICRO, '(XHRRequest)Http.checkConnectivity()', 'Sending; ' + upUrl);
+        Logger.logAction(Logger.LOG_MICRO, '(XHRRequest)Http.checkConnectivity()', 'Sending; ' + internetUpUrl);
         this.doUri(
           HttpMethods.Get,
           null as any,
-          upUrl,
+          internetUpUrl,
           null,
           null,
-          null,
-          function (err?: ErrorInfo | ErrnoException | null, responseText?: unknown) {
-            const result = !err && (responseText as string)?.replace(/\n/, '') == 'yes';
+          internetUpParams,
+          function (
+            err?: ErrorInfo | ErrnoException | null,
+            responseText?: unknown,
+            headers?: any,
+            packed?: boolean,
+            statusCode?: number
+          ) {
+            let result = false;
+            if (!internetUpUrlIsDefault) {
+              result = !err && isSuccessCode(statusCode as number);
+            } else {
+              result = !err && (responseText as string)?.replace(/\n/, '') == 'yes';
+            }
```

Comment:
I think this logic is unnecessary, if we get _any_ HTTP response, then the check is surely complete?

Reply:
Yeah that's a fair point. I don't know the history of why we check the response body every time so I decided to err on the side of caution and preserve the existing behaviour.

Reply:
I suspect the history is that the original implementor thought, well if there's just one URL we control, and we know what we expect the response to be, perhaps we just assert what that response is, but "checking the Internet is available" doesn't require that the URL returns "yes", nor that it returns a success response, simply establishing a TCP connection is enough.

cc @paddybyers in case you have further insight into why we might want to this check to continue to be this specific.

Reply:
There's always the possibility that there's something wrong with the client's DNS resolver, say, where the request resolves to something that responds with an error message, but isn't indicative of there being general connectivity.

That issue would be addressed by insisting that it's an HTTPS URL.